x-pack/legacy/plugins/apm/public/services/rest/callApi.ts

```diff
@@ -55,6 +57,26 @@ export async function callApi<T = void>(
   return res;
 }
 
+export const callApmApi: Client<APMAPI['_S']> = (options => {
+  const { pathname, params = {}, ...opts } = options;
+
+  const path = (params.path || {}) as Record<string, any>;
+  const body = params.body ? JSON.stringify(params.body) : undefined;
+  const query = params.query || {};
+
+  return callApi({
+    ...opts,
+    pathname: Object.keys(path || {}).reduce((acc, paramName) => {
+      if (!(paramName in path)) {
+        throw new Error(`Specified unexpected path parameter ${paramName}`);
```

Comment:
Another advantage of moving from path params to query params. We can at compile time make sure this doesn't happen with query params but not with path params.

Reply:
Hmm, this code is non-sensical. It isn't needed for TS, and `paramName` is guaranteed to be a key of `path`. Perhaps it was a leftover of a different approach, can't remember. I'll remove the check. But as to your point: other than this, are there other examples where path params are less safe than query params?